server/market/orderrouter.go

```diff
@@ -286,17 +297,24 @@ func (r *OrderRouter) handleMarket(user account.AccountID, msg *msgjson.Message)
 		return msgjson.NewError(msgjson.FundingError,
 			fmt.Sprintf("not enough funds. need at least %d, got %d", reqVal, valSum))
 	}
+
+	// Commitment.
+	if len(market.Commit) != order.CommitmentSize {
+		return msgjson.NewError(msgjson.OrderParameterError, "invalid commitment")
+	}
+	var commit order.Commitment
+	copy(commit[:], market.Commit)
+
 	// Create the market order
-	clientTime := encode.UnixTimeMilli(int64(market.ClientTime))
-	serverTime := time.Now().Truncate(time.Millisecond).UTC()
 	mo := &order.MarketOrder{
 		P: order.Prefix{
 			AccountID:  user,
 			BaseAsset:  market.Base,
 			QuoteAsset: market.Quote,
 			OrderType:  order.MarketOrderType,
-			ClientTime: clientTime,
-			ServerTime: serverTime,
+			ClientTime: encode.UnixTimeMilli(int64(market.ClientTime)),
+			//ServerTime set in epoch queue processing pipeline.
```

Comment:
Was surprising to see ServerTime getting set for market and cancel orders here instead of allowing Market.Run to do it, but that's where it is supposed to happen, not before.  It was correct for limit orders.

Reply:
Pretty sure this was done before the details of timestamping had been worked out.

Reply:
For sure!  I should have changed the market and cancel order bits, but apparently I only did the limit orders.